### DISCUSSION TITLE
Investigate providing a download mirror for all release packages

### DISCUSSION THREAD

**Discussion by kilasuit** [PowerShell/PowerShell#20832] [General]
As reported in https://github.com/PowerShell/PowerShell/issues/14589 by @SneWs where they were having what seems to be localised networking issues, it *may* be worthwhile investigating other mirrors to host alI packages that are released for download.

Note: this should be in addition to all current mechanisms of downloading PowerShell, just in case access to the current hosting for release packages (which for all GitHub Releases is AWS) becomes unavailble.
It is additionally noted that this may be a **bigger** than a PowerShell repository/PowerShell Organisation issue and a wider Microsoft/GitHub issue.
The obvious first choice for another mirror service would be a publicly accessible Azure Storage location or additionally could make use of torrent/magnet links.

**Comment by doctordns** [comment]
Given that Microsoft now owns GitHub, isn't the real answer to move GitHub to Azure and use Azure's CDN? ;-)

I would not want to ask thh PowerShell team to have to undertake the work to do the mirroring and deal with day to day monitoring and troubleshooting.
After all, isn't that what all this cloud stuff is for?

**Comment by microsoft-github-policy-service** [comment]
This issue has not had any activity in 6 months, if this is a bug please try to reproduce on the latest version of PowerShell and reopen a new issue and reference this issue if this is still a blocker for you.

**Comment by microsoft-github-policy-service** [comment]
This issue has not had any activity in 6 months, if this is a bug please try to reproduce on the latest version of PowerShell and reopen a new issue and reference this issue if this is still a blocker for you.

**Comment by microsoft-github-policy-service** [comment]
This issue has not had any activity in 6 months, if this is a bug please try to reproduce on the latest version of PowerShell and reopen a new issue and reference this issue if this is still a blocker for you.

**Comment by kilasuit** [comment]
> Given that Microsoft now owns GitHub, isn't the real answer to move GitHub to Azure and use Azure's CDN? ;-)
> 
> I would not want to ask thh PowerShell team to have to undertake the work to do the mirroring and deal with day to day monitoring and troubleshooting. After all, isn't that what all this cloud stuff is for?

Good points, however that wouldn't complete the ask here as it's just moving from one provider to another & getting the same service, whereas the ask was for giving an additional way/s to download releases.

Also I edited the issue as I think there is value in providing download mechanisms like torrents/magnet links for this in future like Canonical do with Ubuntu iso's

**Comment by SteveL-MSFT** [comment]
As part of https://github.com/PowerShell/PowerShell-RFC/blob/master/Draft-Accepted/RFC0065-Install-PowerShell.md which is being implemented, it will use an Azure blob instead of going through GitHub so that should alleviate the original concern.  In addition, since this issue was first opened, we haven't seen additional feedback on throttling of the downloads and we have made the releases through more means including Microsoft Update.
